### ISSUE TITLE
[Feature request]: Use geospatial networks (OSM, GIS) to model traffic on this network

### ISSUE THREAD

**Issue by jdoes12** [projectmesa/mesa#586]
**What's the problem this feature will solve?**
Importing a network from QGIS or OSM and using the examples as "Boltzmann_wealth_model_network" or "virus_on_network" as template for a transportion network (cars moving from node to node) gives distorted network when visualizing it. The NetworkModule does not have a library for plotting geospational coordinates or xy coordinates properly, so the network gets distorted.

When I create a Graph with the following code, then importing it in one of the examples, the visualisation gets distorted, because all the nodes still have coordinates in latlon. Also when they are converted into UTM or XY coordinates.

`G = nx.read_shp('path/shapefile.shp',
    simplify=True)  # use simplify is true otherwise chart get shifted
w_G = nx.Graph()
lat_lon_to_index = {}
for i, node in enumerate(G.nodes()):  # adds nodes with label 1 till n nodes to the network.
    w_G.add_node(i, pos=node)
    lat_lon_to_index[node] = i

for edge in G.edges(
        data=True): 
    w_G.add_edge(lat_lon_to_index[edge[0]], lat_lon_to_index[edge[1]], dis=edge[2]['cost'])`

**Describe the solution you'd like**
Adding the visualisation of mesa in the NetworkModule an option to proper visualise a network from OSM or QGIS when importing it as shapefile.

Hope you can help me with this!

**Additional context**

**Comment by jackiekazil** [comment]
Have you tried https://github.com/Corvince/mesa-geo?

**Comment by ewirth** [comment]
That branch looks dead, those geospatial file supports should be included to the main project. GeoTiff raster support would also earn a lot.

**Comment by jackiekazil** [comment]
@ewirth the repo is alive. We don't included in the main project for complexity reasons. Geospatial is heavy and not everyone needs it.

**Comment by ewirth** [comment]
Hmm, I think maybe you should include it in some form the main project (like extension functions).
Geospatial stuff would definitely stimulate users to use mesa (e.g., not GAMA-ABM).
I would say only two tutorials would be enough:
- Traffic simulation on city, country roads, where you read in Shapefile or GeoJSON (even better) vector data
- And something other, e.g., flow simulation, where you use GeoTiff raster data (like a DEM: Digital Elevation Model)

Basically the Geospatial community is Pythonist, so instead of picking NetLogo, Repast, GAMA-ABM or something Java, they would use Mesa.

**Comment by Corvince** [comment]
@ewirth I am the author of mesa-geo. Development has indeed stalled a bit, but the project is open for any contributions. There had been plans in the past to include mesa-geo under the projectmesa GitHub organization for greater visibility, but since it is still missing tests and a better documentation this has not happened yet. However, the python GIS stack is indeed quite heavy and a bit cumbersome to install (at least on windows without conda). Therefore it is very unlikely that we will include it in the core library. What do you think is the disadvantage of having a separate package?

The traffic simulation example should already be doable with mesa-geo, raster data is not yet supported. But as I said, contributions are always encouraged

**Comment by ewirth** [comment]
@Corvince @jackiekazil 
I would say, it's better to integrate it. Since now, if new people visiting the package would have easily thought that the geospatial support of the package is weak, it seems there is a two years fork without very active development. I suggest integrating it with some author contributions, e.g., who found those extension. 

As you said, you are total right, the GDAL, OGR, Proj4 Python package is heavy, and its mostly maintained in C.
But those things are mostly about data conversions, which aren't necessary. I suggest that you (or if I can help) may only focus on the main geospatial formats (with GDAL anybody can convert their data to a proper file format):
- So in the vector data, I think GeoJSON is the most popular open format (easy to handle in web, any projection supported), or the 'half-proprietary' binary ESRI Shapefile
- In raster data, I would say the same: the most popular is GeoTIFF, we mesa could read GeoTiff, it would also magnet a lot of users.
- And then comes the 3D, maybe for evacuation simulation, e.g., CityGML, or IndoorGML (but I would focus on the first two).

You know there is Go (Golang), which can threaten Python in the future like (10-20 years), but Python still rocks, since Go lacks a lot of libraries, community, and syntax maybe isn't that friendly like Python.

**Comment by tpike3** [comment]
Closing due to mesa-geo progress
